src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java

```diff
@@ -0,0 +1,345 @@
+            0.001,
+            ROLLERS_SENSOR_TO_MECHANISM_RATIO);
+    this.leftRollerSim =
+        new VelocitySystemSim(
+            leftRollerMotor,
+            IntakeConstants.ROLLERS_MOTOR_INVERTED,
+            0.02,
+            0.001,
+            ROLLERS_SENSOR_TO_MECHANISM_RATIO);
+    this.drumMotorSim =
+        new VelocitySystemSim(drumMotor, IntakeConstants.DRUM_MOTOR_INVERTED, 1.0, 0.3, 5.0 / 2.0);
+    this.kickerMotorSim =
+        new VelocitySystemSim(kickerMotor, IntakeConstants.KICKER_MOTOR_INVERTED, 1.0, 0.3, 1.0);
+  }
+
+  @Override
+  public void updateInputs(IntakeIOInputs inputs) {
+    this.rightRollerSim.updateSim();
+    this.leftRollerSim.updateSim();
+    this.drumMotorSim.updateSim();
+    this.kickerMotorSim.updateSim();
+
+    BaseStatusSignal.refreshAll(
+        rightRollerVelocityStatusSignal,
+        leftRollerVelocityStatusSignal,
+        drumVelocityStatusSignal,
+        rightRollerStatorCurrentStatusSignal,
+        leftRollerStatorCurrentStatusSignal,
+        drumStatorCurrentStatusSignal,
+        rightRollerSupplyCurrentStatusSignal,
+        leftRollerSupplyCurrentStatusSignal,
+        drumSupplyCurrentStatusSignal,
+        kickerSupplyCurrentStatusSignal,
+        kickerStatorCurrentStatusSignal,
+        kickerVelocityStatusSignal);
+
+    // FIXME: TEMPORARILY added ! to the IRSensor.get() as it was contradictory on the actual intake
+    inputs.isRightRollerIRBlocked = !rightRollerIRSensor.get();
+    inputs.isLeftRollerIRBlocked = !leftRollerIRSensor.get();
+    inputs.isDrumIRBlocked = !drumIRSensor.get();
+    inputs.isKickerIRBlocked = !kickerIRSensor.get();
+
+    inputs.rightRollerStatorCurrentAmps = rightRollerStatorCurrentStatusSignal.getValueAsDouble();
+    inputs.leftRollerStatorCurrentAmps = leftRollerStatorCurrentStatusSignal.getValueAsDouble();
+    inputs.drumStatorCurrentAmps = drumStatorCurrentStatusSignal.getValueAsDouble();
+    inputs.kickerStatorCurrentAmps = kickerStatorCurrentStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerSupplyCurrentAmps = rightRollerSupplyCurrentStatusSignal.getValueAsDouble();
+    inputs.leftRollerSupplyCurrentAmps = leftRollerSupplyCurrentStatusSignal.getValueAsDouble();
+    inputs.drumSupplyCurrentAmps = drumSupplyCurrentStatusSignal.getValueAsDouble();
+    inputs.kickerSupplyCurrentAmps = kickerSupplyCurrentStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerVelocityRotationsPerSecond =
+        rightRollerVelocityStatusSignal.getValueAsDouble();
+    inputs.leftRollerVelocityRotationsPerSecond = leftRollerVelocityStatusSignal.getValueAsDouble();
+    inputs.drumVelocityRotationsPerSecond = drumVelocityStatusSignal.getValueAsDouble();
+    inputs.kickerVelocityRotationsPerSecond = kickerVelocityStatusSignal.getValueAsDouble();
+
+    // what method would be used to get the value of the reference velocity as a status signal?
+    // could not find any methods that returned a status signal for the reference velocity
```

Comment:
getClosedLoopReference is the method; you can use that and add instance variables for this status signal like the others (and refresh it in the above code)